arangod/MMFiles/MMFilesCompactionFeature.cpp

```diff
@@ -0,0 +1,128 @@
+    _maxFiles(3),
+    _maxSizeFactor(3),
+    _smallDatafileSize(128 * 1024),
+    _maxResultFilesize(128 * 1024 * 1024),
+    _deadNumberThreshold(16384),
+    _deadSizeThreshold(128 * 1024),
+    _deadShare(0.1) { 
+  setOptional(true);
+  requiresElevatedPrivileges(false);
+  onlyEnabledWith("MMFilesEngine");
+  MMFilesCompactionFeature::COMPACTOR = this;
+}
+
+
+void MMFilesCompactionFeature::collectOptions(std::shared_ptr<options::ProgramOptions> options) {
+  options->addSection("compaction", "Configure the MMFiles compactor thread");
+  
+  options->addOption("--compaction.sleep-time", "sleep interval between two compaction runs (in s)",
+                     new UInt64Parameter(&_compactionSleepTime));
+  
+  options->addOption("--compaction.min-interval", "minimum sleep time between two compaction runs (in s)",
+                     new DoubleParameter(&_compactionCollectionInterval));
+
+
+  options->addOption("--compaction.min-small-data-file-size", "minimal filesize threshhold original data files have to be below for a compaction",
+                     new UInt64Parameter(&_smallDatafileSize));
+  
+  options->addOption("--compaction.dead-documents-threshold", "how many dead documents should datafiles to compact contain at least",
+                     new UInt64Parameter(&_deadNumberThreshold));
+  
+  options->addOption("--compaction.dead-size-threshold", "how many bytes of the source datafile should be unused at least",
+                     new UInt64Parameter(&_deadSizeThreshold));
+  
+  options->addOption("--compaction.dead-size-percent-threshold", "how many percent of the source datafile should be unused at least",
+                     new DoubleParameter(&_deadShare));
+
+
+  options->addOption("--compaction.max-files", "maximum number of files to compact to one file",
+                     new UInt64Parameter(&_maxFiles));
+
+  options->addOption("--compaction.max-result-file-size", "how large may the compaction result file become (in bytes)",
+                     new UInt64Parameter(&_maxResultFilesize));
+
+  options->addOption("--compaction.max-file-size-factor", "how large the resulting compacted file may be in comparison to the original",
+                     new UInt64Parameter(&_maxSizeFactor));
+}
+
+void MMFilesCompactionFeature::validateOptions(std::shared_ptr<options::ProgramOptions> options) {
+  if (_deadNumberThreshold < 1024) {
+    LOG_TOPIC(WARN, Logger::COMPACTOR)
+      << "compaction.dead-documents-threshold should be at least 1024.";
+    _deadNumberThreshold = 1024;
+  }
+  
+  if (_deadSizeThreshold < 10240) {
+    LOG_TOPIC(WARN, Logger::COMPACTOR)
+      << "compaction.dead-size-threshold should be at least 10k.";
+    _deadNumberThreshold = 10240;
+  }
+
+  if (_deadShare < 0.001) {
+    LOG_TOPIC(WARN, Logger::COMPACTOR)
+      << "compaction.dead-size-percent-threshold should be at least 0.01%.";
+    _deadShare = 0.01;
+  }
+
+  if (_maxFiles < 2) {
```

Comment:
Why don't we allow a value of 1? That would allow users to turn off merging of adjacent datafiles. Though not ideal, I don't see any reason not to allow this. Is there one?